# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.cpp
# Commit: a71a2a98b6bc
# Full Hash: a71a2a98b6bc017a7bb60d4b7c41f543f156eede
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r d52858ae58d2 -r a71a2a98b6bc dom/script/ScriptSettings.cpp
--- a/dom/script/ScriptSettings.cpp	Mon Feb 21 15:39:04 2022 +0000
+++ b/dom/script/ScriptSettings.cpp	Mon Feb 21 15:39:05 2022 +0000
@@ -57,13 +57,9 @@
 
   LoadedScript* script = static_cast<LoadedScript*>(aPrivateValue.toPrivate());
 
-  if (!script->GetFetchOptions()) {
-    return nullptr;
-  }
-
   JS::Rooted<JS::Value> elementValue(aCx);
   {
-    nsCOMPtr<Element> domElement = script->GetFetchOptions()->mElement;
+    nsCOMPtr<Element> domElement = script->GetScriptElement();
     if (!domElement) {
       return nullptr;
     }
